pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/ast/AbstractApexNode.java

```diff
@@ -144,4 +146,17 @@ public String getNamespace() {
         // TODO(b/239648780)
         return null;
     }
+
+    /** Returns the string value of the {@link Expression expr}. */
+    protected static String expressionToString(Expression expr) {
+        if (expr instanceof LiteralExpression) {
+            if (expr instanceof LiteralExpression.StringVal) {
+                return ((LiteralExpression.StringVal) expr).getValue();
+            } else if (expr instanceof LiteralExpression.NullVal) {
+                return "";
+            }
+            return ((LiteralExpression) expr).asCodeString();
+        }
+        return null;
```

Comment:
Prefer `Optional<String>` over `@Nullable String` over `String`

go/java-practices/null

Reply:
This style (`return null`) is used all over PMD, so I got used to it, but since this is my own unrelated method, I will use `Optional`. Thank you